lib/offline/storage.js

```diff
@@ -537,6 +537,27 @@ shaka.offline.Storage = class {
     // Let the application choose which tracks to store.
     const chosenTracks =
         await config.offline.trackSelectionCallback(allTracks);
+    const duration = manifest.presentationTimeline.getDuration();
+    let sizeEstimate = 0;
+    for (const track of chosenTracks) {
+      const trackSize = track.bandwidth * duration / 8;
+      sizeEstimate += trackSize;
+    }
+    try {
+      const allowedDownload =
+          await config.offline.downloadSizeCallback(sizeEstimate);
+      if (!allowedDownload) {
+        throw new shaka.util.Error(
+            shaka.util.Error.Severity.CRITICAL,
+            shaka.util.Error.Category.STORAGE,
+            shaka.util.Error.Code.OPERATION_ABORTED);
+      }
+    } catch (e) {
+      throw new shaka.util.Error(
+          shaka.util.Error.Severity.CRITICAL,
+          shaka.util.Error.Category.STORAGE,
+          shaka.util.Error.Code.OPERATION_ABORTED);
```

Comment:
This is what happens if the callback throws, correct?  I think in that case, we should at least log something, and maybe even use a different error code to make it clear what happened.

Reply:
There are two causes here, the callback is asynchronous and it is I hope it returns a boolean, if it is false we should throw a "storage limit reached" error, in the case that an error is thrown (throw) I think the aborted operation is reasonable, what do you think?

Reply:
I agree with Joey, there should be at least a log, preferably a different error.  Returning `false` already aborts the operation since we throw the STORAGE_LIMIT_REACHED error.  This is more likely to happen if the app typed something wrong and we got a TypeError or something.

Reply:
Ok, I will add a new error called `DOWNLOAD_SIZE_CALLBACK_ERROR`

Reply:
Done. Can you review again?